joomla/joomla-cms
# [4] Clearer code for urls_position (com_content article default.php)

Closes https://github.com/joomla/joomla-cms/issues/32623

### Summary of Changes
- Simplified links display code in `article/default.php`.
- Added additional option "Don't display"/`JHIDE` to setting `urls_position` if somebody doesn't want to display links at all (or wants to use the feature elsewhere).
- Removed `$urls = json_decode($this->item->urls);` in `article/.../default.php`.
- Add indents in `components/com_content/tmpl/article/default.xml`

### Testing Instructions
- Optional: If you want  to test the default behavior of a clean Joomla 4 before you apply the patch.
- - Install Joomla 4 nightly.
- - "Install Blog Sample Data" in Control Panel.
- - See an article in front-end. Debug `$params` in `components\com_content\tmpl\article\default.php`. **There is no parameter for `urls_position` in it.**
- - - **That's the reason why I use (int) in PHP code for B\C reasons.**

- Edit article "Welcome to your blog". Add a link/url in tabulator `Images and Links`. Save article.
- - See comment below if you don't see that tabulator: https://github.com/joomla/joomla-cms/pull/32713#issuecomment-801489385
- Link is displayed `Above` (inherited global setting) in front-end.

- Apply patch
- - See article in front-end. Link is still displayed above (global setting).
- - Play around with the setting  `Positioning of the Links`  in
- - Article edit.
- - `administrator/index.php?option=com_config&view=component&component=com_content`

### Documentation Changes Required
I don't know. New setting "Hide".
